src/main/java/uk/gov/justice/hmpps/prison/service/enteringandleaving/ReleasePrisonerService.kt

```diff
@@ -34,15 +33,33 @@ class ReleasePrisonerService(
     val booking = getOffenderBooking(prisonerIdentifier).flatMap { it.isActiveIn() }.getOrThrow()
     val toLocation = getAgencyLocation(request.toLocationCode).getOrThrow()
 
-    val movement = externalMovementService.updateMovementsForRelease(request.releaseTime, request.movementReasonCode, booking, toLocation, request.commentText)
+    val movement = externalMovementService.updateMovementsForRelease(
+      request.releaseTime,
+      request.movementReasonCode,
+      booking,
+      toLocation,
+      request.commentText,
+    )
 
-    caseNoteMovementService.createReleaseNote(booking, movement)
-    bedAssignmentMovementService.endBedHistory(booking.bookingId, movement.movementTime)
-    sentenceMovementService.deactivateSentences(booking.bookingId)
-    paymentsMovementService.endPaymentRules(booking.bookingId, movement.movementDate)
-    activityMovementService.endActivitiesAndWaitlist(booking, movement.fromAgency, movement.movementDate, request.movementReasonCode)
-
-    return booking.release(toLocation, movement.movementTime, movement.movementReason.code)
+    return booking
+      .apply {
+        inOutStatus = "OUT"
+        isActive = false
+        bookingStatus = "C"
+        livingUnitMv = null
+        assignedLivingUnit = null
+        location = toLocation
+        bookingEndDate = movement.movementTime
+        statusReason = "REL-${movement.movementReason.code}"
+        commStatus = null
+      }
+      .also {
+        caseNoteMovementService.createReleaseNote(it, movement)
+        bedAssignmentMovementService.endBedHistory(it.bookingId, movement.movementTime)
+        sentenceMovementService.deactivateSentences(it.bookingId)
+        paymentsMovementService.endPaymentRules(it.bookingId, movement.movementDate)
+        activityMovementService.endActivitiesAndWaitlist(it, movement.fromAgency, movement.movementDate, request.movementReasonCode)
+      }
       .let { transformer.transform(it) }
```

Comment:
I decided that putting the booking mutation front and centre means I have an instant overview of what happens during a release.